### PR TITLE
Fix race in data type `Object`

### PR DESCRIPTION
**Pull request by CurtizJ**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by CurtizJ** [comment]
The problem was that simdjson parser is not thread safe, so we cannot share ptr to serialization.

**Review by alexey-milovidov — COMMENTED** [review]
We need some general approach like:

1. ISerialization is presumed to be not thread safe, so you cannot reuse it in multiple threads *for any data types*.
Also you can highlight it explicitly by making its method non-const (not sure if it is possible)...

2. Or we can state that it is thread safe, but for simdjson, specifically, we will maintain an ObjectPool inside. This is similar to what we do for `re2_st`.

**Comment by alexey-milovidov** [comment]
Also need to check why this problem was not covered by stress tests...
Maybe we need to add more randomization?

**Comment by CurtizJ** [comment]
Maybe we can just recreate parser on each row, because it's a quite light object and we don't use any functionality like `reserve` or smth.

**Comment by alexey-milovidov** [comment]
It's at least one extra virtual call and allocation (unique_ptr).

**Comment by CurtizJ** [comment]
@alexey-milovidov 

I've added a pool of parsers and optimized performance of inserts a bit more.

File: src/Columns/ColumnObject.cpp [hunk 1]
```diff
@@ -284,29 +307,26 @@ void ColumnObject::Subcolumn::insert(Field field, FieldInfo info)
     if (!is_nullable && info.have_nulls)
         field = applyVisitor(FieldVisitorReplaceNull(base_type->getDefault(), value_dim), std::move(field));
 
-    auto value_type = createArrayOfType(base_type, value_dim);
```
Comment (CurtizJ) [inline src/Columns/ColumnObject.cpp:287]: Here was one of the performance issues, because many excessive `DataTypePtr` object were created.

**Review by alexey-milovidov — APPROVED** [review]

**Pull request merged**